js/js.libraries/src/core/collections.kt

```diff
@@ -92,6 +92,27 @@ public fun <T> MutableList<T>.fill(value: T): Unit {
 }
 
 /**
+ * Randomly shuffles elements in this list.
+ *
+ * See: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+ */
+@SinceKotlin("1.2")
+public fun <T> MutableList<T>.shuffle(): Unit {
+    (lastIndex downTo 1).forEach { i ->
+        rand(i + 1).let { j ->
+            swap(i, j)
+        }
+    }
+}
+private fun rand(upperBound: Int) = Math.floor(Math.random() * upperBound)
```

Comment:
Couldn't find any way to make `Random` customizable in `JavaScript`, therefore that variant is left out.

Reply:
We're going to provide common `Random` , see https://youtrack.jetbrains.com/issue/KT-17261, though it isn't there yet.